src/engine/Story.ts

```diff
@@ -387,16 +431,59 @@ export class Story extends InkObject {
       }
 
       this.state.didSafeExit = false;
+      this._sawLookaheadUnsafeFunctionAfterNewline = false;
 
       if (this._recursiveContinueCount == 1)
         this._state.variablesState.batchObservingVariableChanges = false;
 
       this._asyncContinueActive = false;
+      if (this.onDidContinue !== null) this.onDidContinue;
     }
 
     this._recursiveContinueCount--;
 
     if (this._profiler != null) this._profiler.PostContinue();
+
+    if (this.state.hasError || this.state.hasWarning) {
+      if (this.onError !== null) {
+        if (this.state.hasError) {
+          for (let err of this.state.currentErrors) {
+            this.onError(err, ErrorType.Error);
+          }
+        }
+        if (this.state.hasWarning) {
+          for (let err of this.state.currentWarnings) {
+            this.onError(err, ErrorType.Warning);
+          }
+        }
+        this.ResetErrors();
+      } else {
+        let sb = new StringBuilder();
+        sb.Append("Ink had ");
+        if (this.state.hasError) {
+          sb.Append(this.state.currentErrors.length);
+          sb.Append(this.state.currentErrors.length == 1 ? " error" : "errors");
+          if (this.state.hasWarning) sb.Append(" and ");
+        }
+        if (this.state.hasWarning) {
+          sb.Append(this.state.currentWarnings.length);
+          sb.Append(
+            this.state.currentWarnings.length == 1 ? " warning" : "warnings"
+          );
+          if (this.state.hasWarning) sb.Append(" and ");
```

Comment:
I think that's an extra line that isn't there upstream :)

Reply:
Haha bad copy/paste I suppose!